vars/unitTestPost.groovy

```diff
@@ -44,46 +50,53 @@ def call(Map config = [:]) {
                      allowEmptyArchive: true
     return
   }
+  
+  if (stage_info['with_valgrind'] == 'disabled') {
+    def test_results = config.get('testResults', 'test_results/*.xml')
+    junit testResults: test_results
+  }
 
-  junit testResults: config.get('testResults', 'test_results/*.xml')
-
-  def artifact_list = config.get('artifacts', ['run_test.sh/*', 'vm_test/**'])
+  def artifact_list = stage_info['artifacts']
   artifact_list.each {
     archiveArtifacts artifacts: it
   }
 
-  publishValgrind failBuildOnInvalidReports: true,
-                  failBuildOnMissingReports: true,
-                  failThresholdDefinitelyLost: '0',
-                  failThresholdInvalidReadWrite: '0',
-                  failThresholdTotal: '0',
-                  pattern: 'dnt.*.memcheck.xml',
-                  publishResultsForAbortedBuilds: false,
-                  publishResultsForFailedBuilds: true,
-                  sourceSubstitutionPaths: '',
-                  unstableThresholdDefinitelyLost: '0',
-                  unstableThresholdInvalidReadWrite: '0',
-                  unstableThresholdTotal: '0'
+  if (stage_info['with_valgrind'] == 'memcheck') {
+    publishValgrind failBuildOnInvalidReports: true,
+                    failBuildOnMissingReports: true,
+                    failThresholdDefinitelyLost: '0',
+                    failThresholdInvalidReadWrite: '0',
+                    failThresholdTotal: '0',
+                    pattern: stage_info['valgrind_pattern'],
+                    publishResultsForAbortedBuilds: false,
+                    publishResultsForFailedBuilds: true,
+                    sourceSubstitutionPaths: '',
+                    unstableThresholdDefinitelyLost: '0',
+                    unstableThresholdInvalidReadWrite: '0',
+                    unstableThresholdTotal: '0'
+  }
+  if (stage_info['with_valgrind'] == 'disabled') {
```

Comment:
This is an `else` isn't it?
```suggestion
  } else {
```

Reply:
I could use an else if. There's a task about enabling pmemcheck some time in the future, so I don't want to assume that anything else should do this code.